### PR TITLE
SDCORE-788: Adding support for N/W initiated subscriber deregistration

### PR DESCRIPTION
**Pull request by gajjarvini**

### CODE REVIEW

File: gmm/sm.go [hunk 1]
```diff
@@ -126,6 +127,10 @@ func Registered(state *fsm.State, event fsm.EventType, args fsm.ArgsType) {
 		logger.GmmLog.Debugln(event)
 	case InitDeregistrationEvent:
 		logger.GmmLog.Debugln(event)
+	case NwInitiatedDeregistrationEvent:
```
Comment (vthiruveedula) [inline gmm/sm.go:130]: when does amf send this event? i haven't seen code to send this event in this PR.

File: gmm/handler.go [hunk 2]
```diff
@@ -2320,7 +2370,11 @@ func HandleDeregistrationAccept(ue *context.AmfUe, anType models.AccessType,
 	}
 
 	ue.DeregistrationTargetAccessType = 0
-	return nil
+
+	return GmmFSM.SendEvent(ue.State[models.AccessType__3_GPP_ACCESS], DeregistrationAcceptEvent, fsm.ArgsType{
```
Comment (vthiruveedula) [inline gmm/handler.go:2374]: why are we sending DeregistrationAcceptEvent? can u please add some comments before the return statement

File: gmm/handler.go [hunk 3]
```diff
@@ -1530,6 +1530,56 @@ func AuthenticationProcedure(ue *context.AmfUe, accessType models.AccessType) (b
 	return false, nil
 }
 
+func NetworkInitiatedDeregistrationProcedure(ue *context.AmfUe,
+	accessType models.AccessType) (err error) {
+
+	anType := util.AnTypeToNas(accessType)
+	if ue.CmConnect(accessType) {
```
[the remark this answers is not in the document]
Comment (thakurajayL) [reply gmm/handler.go:1537]: No paging in this phase. If subscriber is idle then just core network cleanup.

[the remark this answers is not in the document]
Comment (vthiruveedula) [reply gmm/handler.go:1537]: ok, cleanup amfUeContext if UE is idle state at the end of this function.

**Pull request closed without merging**